device/alpaka/src/utils/queue.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/alpaka/utils/queue.hpp"
+
+#include "utils.hpp"
+
+// Alpaka include(s).
+#include <alpaka/alpaka.hpp>
+
+namespace traccc::alpaka {
+
+struct queue::impl {
+
+    /// Constructor
+    /// @param device The device to create the queue for
+    explicit impl(std::size_t device)
+        : m_device(device == INVALID_DEVICE ? 0 : device),
+          m_queue(::alpaka::getDevByIdx(::alpaka::Platform<Acc>{}, m_device)) {}
+
+    /// The device the queue is created for
+    std::size_t m_device;
+
+    /// The real Alpaka queue object
+    Queue m_queue;
+
+};  // struct queue::impl
+
+queue::queue(std::size_t device) : m_impl{std::make_unique<impl>(device)} {}
+
+queue::queue(void* input_queue)
+    : m_impl{std::make_unique<impl>(INVALID_DEVICE)} {
+
+    // Copy the queue pointer
+    m_impl->m_queue = *static_cast<Queue*>(input_queue);
```

Comment:
:thinking: This is not exactly how I imagined it...

This interface, with a `void*` pointer suggests that it just wraps around an existing `alpaka::Queue`. Not make a copy of it, and then use the copy from there on.

I rather imagined making the `impl` struct something like this:

https://github.com/acts-project/vecmem/blob/main/sycl/src/utils/sycl/queue_wrapper.sycl#L20-L25

**Not** with a `shared_ptr`, but rather with a `unique_ptr`. (Since this class, unlike `vecmem::sycl::queue_wrapper`, cannot be copied.) So the main thing is that there would be both a bare pointer, and a unique pointer inside of `impl`. Which would be set up according to how the main class was created.